Sources/Sentry/SentryProfiler.mm

```diff
@@ -428,12 +428,16 @@ + (void)updateProfilePayload:(NSMutableDictionary<NSString *, id> *)payload
               forTransaction:(SentryTransaction *)transaction;
 {
     payload[@"platform"] = transaction.platform;
+
+    const auto activeThreadID =
+        [transaction.trace.transactionContext sentry_threadInfo].threadId ?: @(-1);
     payload[@"transaction"] = @{
         @"id" : transaction.eventId.sentryIdString,
         @"trace_id" : transaction.trace.traceId.sentryIdString,
         @"name" : transaction.transaction,
-        @"active_thread_id" : [transaction.trace.transactionContext sentry_threadInfo].threadId
+        @"active_thread_id" : activeThreadID
```

Comment:
I'm not sure what the right thing to put here would be.

Reply:
Do you mean if the thread ID is NULL? If yes, couldn't we just not set the `active_thread_id`?

Reply:
@phacops Please let us know how we would proceed in this case. Maybe we just choose not to send the profile at all.